scriptler/addShellPreBuildStepToProject.groovy

```diff
@@ -0,0 +1,34 @@
+/*** BEGIN META {
+ "name" : "Add a pre-build shell script step",
+ "comment" : "Add a pre-build shell script step to all Maven and Frestyle Projects",
+ "core": "1.609",
+ "authors" : [
+ { name : "Kuisathaverat" }
+ ]
+ } END META**/
+
+import hudson.util.*
+import hudson.tasks.*
+import hudson.maven.*
+
+
+Jenkins.instance.getAllItems(Job.class)
+	.findAll{ it instanceof FreeStyleProject || it instanceof MavenModuleSet  }
+	.each{
+      println it.name + " - " + it.class
+
+      DescribableList<Builder,Descriptor<Builder>> builders = new DescribableList<Builder,Descriptor<Builder>>()
+      builders.add(new hudson.tasks.Shell("echo 'Hello world'"))
```

Comment:
:bug: Given that this is pretty clearly just an example, I'm not sure we want it in the Scriptler catalog as is - if you want to make it just do shell steps and have the shell script as a parameter, that'd work.